Multiply TextEdit line spacing by the editor scale

Calinou opened the pull request:
This makes sure the default line spacing in the script editor is consistent with the editor scale in use.

**Before (editor scale set to 200%, line spacing set to 6):**
![textedit_before_edscale_fix](https://user-images.githubusercontent.com/180032/43288418-0c2251f0-9128-11e8-8375-50e907aa2155.png)

**After (editor scale set to 200%, line spacing set to 6):**
![textedit_after_edscale_fix](https://user-images.githubusercontent.com/180032/43288408-03cce06a-9128-11e8-9743-8b72bd885b12.png)

akien-mga commented on scene/gui/text_edit.cpp:
```diff
@@ -879,7 +879,11 @@ void TextEdit::_notification(int p_what) {
 					}
 
+#ifdef TOOLS_ENABLED
+					int ofs_y = (i * get_row_height() + cache.line_spacing * EDSCALE / 2) + ofs_readonly;
```
It would be cleaner to do this only once where `cache.line_spacing` is defined.

Calinou committed:
Multiply TextEdit line spacing by the editor scale

This makes sure the default line spacing in the script editor
is consistent with the editor scale in use.

akien-mga commented:
Now that Godot 3.1 has been released, we don't plan to cherry-pick new features and enhancements to the 3.0 branch, unless there is very strong support in favor of it. Removing cherrypick label for 3.0.
